File: passenger-service/src/test/java/io/voitovich/yura/passengerservice/unit/util/UnitTestsUtils.java
package io.voitovich.yura.passengerservice.unit.util;

import io.voitovich.yura.passengerservice.dto.request.PassengerProfilePageRequest;
import io.voitovich.yura.passengerservice.dto.request.PassengerProfileUpdateRequest;
import io.voitovich.yura.passengerservice.dto.request.PassengerSaveProfileRequest;
import io.voitovich.yura.passengerservice.dto.response.PassengerProfilePageResponse;
import io.voitovich.yura.passengerservice.entity.PassengerProfile;
import io.voitovich.yura.passengerservice.event.model.ReceiveRatingModel;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public class UnitTestsUtils {


    public static PassengerProfile createDefaultPassengerProfileWithoutId() {
        return PassengerProfile.builder()
                .surname("surname")
                .name("name")
                .phoneNumber("+375295432551")
                .build();
    }

    public static PassengerProfilePageResponse createDefaultPassengerProfilePageResponse() {
        return PassengerProfilePageResponse.builder()
                .profiles(List.of())
                .totalPages(1)
                .pageNumber(1)
                .totalElements(0)
                .build();
    }

    public static PassengerProfilePageRequest createDefaultPassengerProfilePageRequest() {
        return PassengerProfilePageRequest.builder()
                .pageNumber(1)
                .pageSize(1)
                .orderBy("id")
                .build();
    }


    public static ReceiveRatingModel createDefaultReceiveRatingModel() {
        return ReceiveRatingModel
                .builder()
                .ratedId(UUID.randomUUID())
                .raterId(UUID.randomUUID())
                .rideId(UUID.randomUUID())
                .rating(BigDecimal.valueOf(5))
                .build();
    }

    public static PassengerSaveProfileRequest createDefaultPassengerProfileSaveRequest() {
        return PassengerSaveProfileRequest
                .builder()
                .phoneNumber("+375295432551")
                .name("name")
                .surname("surname")
                .build();
    }


    public static PassengerProfile createDefaultPassengerProfileWithId(UUID uuid) {
        return PassengerProfile.builder()
                .id(uuid)
                .surname("Surname")
                .name("name")
                .phoneNumber("+375295432551")
                .build();
    }

    public static PassengerProfileUpdateRequest createDefaultPassengerProfileUpdateRequest(UUID uuid) {
        return PassengerProfileUpdateRequest
                .builder()
                .id(uuid)
                .surname("surname")
                .name("name")
                .phoneNumber("+375295432551")
                .build();
    }
}

File: ride-service/src/test/kotlin/io/voitovich/yura/rideservice/unit/util/UnitTestsUtils.kt
package io.voitovich.yura.rideservice.unit.util

import io.voitovich.yura.rideservice.client.model.PassengerProfileModel
import io.voitovich.yura.rideservice.dto.mapper.RideMapper
import io.voitovich.yura.rideservice.dto.request.*
import io.voitovich.yura.rideservice.event.model.ConfirmRatingReceiveModel
import org.locationtech.jts.geom.Point
import java.math.BigDecimal
import java.util.*

class UnitTestsUtils {

    companion object {
        fun createDefaultGetAvailableRidesRequest(): GetAvailableRidesRequest {
            return GetAvailableRidesRequest(id = UUID.randomUUID(), currentLocation = createLocation(), radius = 300)
        }

        fun createDefaultGetAvailableRidesRequestWithoutRadius(): GetAvailableRidesRequest {
            return GetAvailableRidesRequest(id = UUID.randomUUID(), currentLocation = createLocation(), radius = null)
        }

        fun createDefaultGetAvailableRidesRequestWithInvalidRadius(): GetAvailableRidesRequest {
            return GetAvailableRidesRequest(id = UUID.randomUUID(), currentLocation = createLocation(), radius = 100)
        }

        fun createDefaultUpdatePositionRequest(): UpdatePositionRequest {
            return UpdatePositionRequest(rideId = UUID.randomUUID(), id = UUID.randomUUID(), location = createLocation())
        }

        fun createDefaultRidePageRequest(): RidePageRequest {
            return RidePageRequest(pageNumber = 1, pageSize = 1, orderBy = "id")
        }

        fun createDefaultConfirmRatingReceiveModel(): ConfirmRatingReceiveModel {
            return ConfirmRatingReceiveModel(rideId = UUID.randomUUID(), rating = BigDecimal.ONE)
        }

        fun createDefaultCreateRideRequest(): CreateRideRequest {
            return CreateRideRequest(passengerId = UUID.randomUUID(), startGeo = createLocation(), endGeo = createLocation())
        }

        fun createDefaultPassengerProfileModel(): PassengerProfileModel {
            return PassengerProfileModel(
                id = UUID.randomUUID(),
                phoneNumber = "+375295432551",
                name = "Name",
                surname = "Surname",
                rating = BigDecimal(5))
        }

        fun createDefaultCancelRequest(): CancelRequest {
            return CancelRequest(passengerId = UUID.randomUUID(), rideId = UUID.randomUUID())
        }

        fun createDefaultAcceptRideRequest(): AcceptRideRequest {
            return AcceptRideRequest(rideId = UUID.randomUUID(), driverId = UUID.randomUUID(), location = createLocation())
        }

        fun createDefaultPoint(mapper: RideMapper): Point {
            return mapper.fromRequestPointToPoint(createLocation())
        }

        fun createDefaultSendRatingRequest(): SendRatingRequest {
            return SendRatingRequest(rideId = UUID.randomUUID(), rating = BigDecimal.ONE)
        }

        fun createLocation(): RequestPoint {
            return RequestPoint(latitude = BigDecimal.TEN, longitude = BigDecimal.TEN)
        }
    }

}